core/src/main/java/io/crate/concurrent/CompletableFutures.java

```diff
@@ -38,4 +41,11 @@ private CompletableFutures() {
         future.completeExceptionally(t);
         return future;
     }
+
+    public static <T> CompletableFuture<List<T>> allAsList(Collection<? extends CompletableFuture<? extends T>> futures) {
```

Comment:
Since we took this implementation "onboard" please add some tests to document its behaviour